tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js

```diff
@@ -29,46 +29,58 @@ define(['require', 'jquery', 'log', 'backbone', 'smart_wizard', 'siddhiAppSelect
                  * @constructs
                  * @class ExportDialog
                  * @param {Object} options exportContainerModal
-                 * @param {boolean} isDocker  is Docker File Export
+                 * @param {boolean} isExportDockerFlow  is Docker File Export
                  */
-                initialize: function (options, isDocker) {
-                    this.app = options;
-                    this.options = _.cloneDeep(_.get(options.config, 'export_dialog'));
-                    this.exportContainer;
-                    this.isDocker = isDocker;
-                    this.payload = {
+                initialize: function (options, isExportDockerFlow) {
+                    this._options = options;
+                    var exportDialog = _.cloneDeep(_.get(options.config, 'export_dialog'));
+                    this._exportContainer = $(_.get(exportDialog, 'selector')).clone();
+
+                    this._isExportDockerFlow = isExportDockerFlow;
+                    this._payload = {
                         templatedSiddhiApps: [],
                         configuration: '',
                         templatedVariables: [],
                         bundles: [],
                         jars: [],
                         kubernetesConfiguration: ''
                     };
-                    this.appTemplatingModel;
-                    this.configTemplateModel;
-                    this.kubernetesConfigModel;
+                    this._siddhiAppSelector;
+                    this._jarsSelectorDialog;
+                    this._appTemplatingModel;
+                    this._configTemplateModel;
+                    this._kubernetesConfigModel;
                     this._fill_template_value_dialog;
+
+                    var type;
+                    if (isExportDockerFlow) {
+                        type = 'docker';
+                    } else {
+                        type = 'kubernetes';
+                    }
+                    var exportUrl = options.config.baseUrl + "/export?type=" + type;
+                    this._btnExportForm =  $('' +
+                        '<form id="submit-form" method="post" enctype="application/x-www-form-urlencoded" target="export-download" >' +
+                        '<button  type="button" class="btn btn-default hidden" id="export-btn" data-dismiss="modal" >Export</button>' +
+                        '</form>').attr('action', exportUrl);
+
                 },
 
                 show: function () {
-                    this.exportContainer.modal('show');
+                    //todo stop form dismiss when clicked away
```

Comment:
remove todos

Reply:
Open an issue for todo https://github.com/siddhi-io/distribution/issues/377